build-logic/convention/build.gradle.kts

```diff
@@ -18,27 +18,16 @@ tasks.withType<ValidatePlugins>().configureEach {
   enableStricterValidation = true
 }
 
-val javaTarget = libs.versions.javaTarget.get()
-// val jdkVersion = libs.versions.jdkVersion.get()
-//
-// java {
-//   toolchain {
-//     languageVersion.set(JavaLanguageVersion.of(jdkVersion))
-//   }
-// }
-
 tasks {
-  val releaseTarget = javaTarget.toInt()
-
   withType<GroovyCompile>().configureEach {
-    options.release = releaseTarget
+    options.release = libs.versions.javaTarget.map(String::toInt)
   }
   withType<JavaCompile>().configureEach {
-    options.release = releaseTarget
+    options.release = libs.versions.javaTarget.map(String::toInt)
   }
   withType<KotlinCompile>().configureEach {
```

Comment:
@autonomousapps You shouldn't configure KotlinCompile directly, because it doesn't affect Kapt, maybe Ksp tasks for example. Using the top level `kotlin { compilerOptions { ...` wire things correctly.